featuretools/entityset/timedelta.py

```diff
@@ -116,66 +103,45 @@ def make_singular(cls, s):
 
     @classmethod
     def _check_unit_plural(cls, s):
-        if len(s) > 1 and not s.endswith('s'):
+        if len(s) > 2 and not s.endswith('s'):
             return (s + 's').lower()
         elif len(s) > 1:
             return s.lower()
         return s
 
-    def get_name(self):
-        if self.unit == self._generic_unit:
-            return str(self.value)
+    def get_unit_type(self):
+        if self.unit == "o":
+            return None
+        elif self.unit in self._absolute_units:
+            return pd.Timedelta(self.value, self.unit)
         else:
-            unit = self.readable_unit
-            if self.readable_unit == "Weeks":
-                # divide to convert back
-                return "{} {}".format(self.value / self._convert_to_days["w"], unit)
-            if self.value == 1:
-                unit = self.make_singular(unit)
-
-            return "{} {}".format(self.value, unit)
+            unit = self.readable_unit.lower()
+            return relativedelta(**{unit: self.value})
 
-    def __eq__(self, other):
-        if not isinstance(other, Timedelta):
-            return False
+    def get_name(self):
+        unit = self.readable_unit
+        if self.readable_unit == "Weeks":
+            # divide to convert back
+            return "{} {}".format(self.value / self._convert_to_days["w"], unit)
+        if self.value == 1:
+            unit = self.make_singular(unit)
 
-        return (self.value == other.value and
-                self.unit == other.unit and
-                self.entity == other.entity and
-                self.inclusive == other.inclusive)
+        return "{} {}".format(self.value, unit)
 
     @property
     def readable_unit(self):
         if self._original_unit is not None:
             return self._readable_units[self._original_unit]
         return self._readable_units[self.unit]
 
-    def get_pandas_timedelta(self):
-        if self.is_absolute() and self.unit != self._generic_unit:
-            return pd.Timedelta(self.value, self.unit)
-        else:
-            return None
-
-    def view(self, unit):
-        if self.is_absolute() and self.unit != self._generic_unit:
-            return self.get_pandas_timedelta().view(unit)
-        else:
-            return None
-
-    @property
-    def value_in_seconds(self):
-        if self.is_absolute() and self.unit != self._generic_unit:
-            pd_td = self.get_pandas_timedelta()
-            return pd_td.total_seconds()
-        else:
-            return None
+    def get_delta_obj(self):
+        return self.delta_obj
 
     def get_arguments(self):
```

Comment:
this technically changes the schema of feature serialization in the case where a feature had a use previous value, so we need to update the schema version in `featuretools/feature_base/features_serializer.py`.

this point inspired this issue: https://github.com/Featuretools/featuretools/issues/698

Reply:
Changed schema version in most recent version.